src/components/MaterialUI/Inputs/CheckboxGroup.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import CheckboxGroupProps, { isCheckboxGroupProps } from "./CheckboxGroupProps";
+import Checkbox from "./Checkbox";
+import CheckboxProps from "./CheckboxProps";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(theme => ({
+	container: {
+		display: "flex",
+		flexDirection: "column",
+	},
+	checkBoxContainer: {
+		display: "flex",
+		flexDirection: "row",
+	},
+	errorText: {
+		marginTop: theme.spacing(0.5),
+		color: theme.palette.error.main,
+		fontSize: theme.typography.fieldLabelSize,
+		float: "left",
+	},
+}));
+
+const CheckboxGroup = ({ checkboxGroupProps }) => {
+	if (isCheckboxGroupProps(checkboxGroupProps) === false) {
+		throw new TypeError("checkboxGroupProps property is not of type CheckboxGroupProps");
+	}
+
+	const update = checkboxGroupProps?.get(CheckboxGroupProps.propNames.update);
+	const value = checkboxGroupProps?.get(CheckboxGroupProps.propNames.value);
+	const label = checkboxGroupProps?.get(CheckboxGroupProps.propNames.label);
+	const error = checkboxGroupProps?.get(CheckboxGroupProps.propNames.error);
+	const readOnly = checkboxGroupProps?.get(CheckboxGroupProps.propNames.readOnly) || false;
+	const disabled = checkboxGroupProps?.get(CheckboxGroupProps.propNames.disabled) || false;
+	const options = checkboxGroupProps?.get(CheckboxGroupProps.propNames.options) ?? [];
+
+	const handleGroupUpdate = (checked, value, newValue) => {
+		let values = value ? value.split("|") : [];
+		let position = values.indexOf(newValue);
+		if (checked && position === -1) {
+			values.push(newValue);
+		}
+
+		if (!checked && position !== -1) {
+			values.splice(position, 1);
+		}
+		update(values.join("|"));
+	};
+
+	const classes = useStyles();
```

Comment:
better to place hook calls on the top of component so, by this way you will minimize chance that somebody won't notice it and add some conditional return before all hooks are called